src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/RazorMapToDocumentEditsParams.cs

```diff
@@ -14,26 +12,21 @@ namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
     // Note: This type should be kept in sync with the one in Razor.LanguageServer assembly.
     internal class RazorMapToDocumentEditsParams : IEquatable<RazorMapToDocumentEditsParams>
     {
-        public RazorLanguageKind Kind { get; set; }
+        public RazorLanguageKind Kind { get; init; }
 
-        public Uri RazorDocumentUri { get; set; }
+        public required Uri RazorDocumentUri { get; init; }
 
-        public TextEdit[] ProjectedTextEdits { get; set; }
+        public required TextEdit[] ProjectedTextEdits { get; init; }
 
-        public TextEditKind TextEditKind { get; set; }
+        public TextEditKind TextEditKind { get; init; }
 
-        public FormattingOptions FormattingOptions { get; set; }
+        public required FormattingOptions? FormattingOptions { get; init; }
 
         // Everything below this is for testing purposes only.
-        public bool Equals(RazorMapToDocumentEditsParams other)
+        public bool Equals(RazorMapToDocumentEditsParams? other)
         {
-            if (object.ReferenceEquals(this, other))
-            {
-                return true;
-            }
-
```

Comment:
Why did you remove this optimization?

Reply:
`// Everything below this is for testing purposes only.` - I don't think such little optimization matters when running tests. There are several classes with the same structure as this one, e.g. `RazorMapToDocumentRangesParams`, that do not have this check in their `Equals` method, so I just removed it for consistency purposes

Reply:
@DustinCampbell So are you ok with this one?

Reply:
Yes, thanks for the explanation.